src/actions/actions.ts

```diff
@@ -583,6 +589,21 @@ export class DeleteOperatorXVisual extends BaseOperator {
 }
 
 @RegisterAction
+export class UpperCaseOperatorXVisual extends BaseOperator {
+    public keys = ["U"];
+    public modes = [ModeName.Visual, ModeName.VisualLine];
+    
+    public async run(vimState: VimState, start: Position, end: Position): Promise<VimState> {
+      [start, end] = adjustVimSelectionPosition(vimState, start, end);
```

Comment:
(Though if you want you can remove the start/stop switching logic I have in the other operators...)